Dockerfile

```diff
@@ -22,23 +20,26 @@ RUN apk --no-cache add \
     postgresql-dev \
     postgresql-client
 
-# Install Python dependencies
-COPY requirements/*.txt /tmp/requirements/
-# Switch to /tmp to install dependencies outside home dir
-WORKDIR /tmp
-# TODO: Consider a way to install only the "build.txt" deps for production.
-RUN pip install --require-hashes --no-cache-dir -r requirements/$PIP_FILE
-
 WORKDIR /app
 COPY . /app
-RUN chown app:app -R .
-USER app
 
+FROM base AS prod_preinstall
+RUN echo "Installing prod dependencies"
+
+COPY requirements/build.txt ./requirements/
+RUN pip install --require-hashes --no-cache-dir -r requirements/build.txt
+
+FROM base AS dev_preinstall
+RUN echo "Installing dev dependencies"
+
+COPY requirements ./requirements/
+RUN pip install --require-hashes --no-cache-dir -r requirements/all.txt
 RUN DEBUG=False SECRET_KEY=foo ./manage.py collectstatic --noinput -c
 
+FROM ${APP_ENV}_preinstall AS release
```

Comment:
Hey @ wlach,

I added changed the stages to be more explicit, now to build successfully we just need to change the CI to build with the command `docker build -t app:build --target <devapp | release>`

Reply:
Also, as a followup, I tried to use setup a virtual env while setting up the container, but even after activating I was still having issues after removing the required dependencies needed for installing the python binaries. Since we are using an alpine image what I decided to do instead was to set the dependencies to virtual dependencies and then uninstall them after we built and installed the required binaries. As of now I am only deleting postgresql-dev be we can alias any of the build dependencies and delete them in one step .